File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Request user for input
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // If statements for before grade one and 16+
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1; // Start with 1 to count the first word
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}

File: Problem Set 6/readability.py
from cs50 import get_string

text = get_string("Text: ")
countOfletters = 0
countOfwords = 1
countOfsentences = 0
for i in range(len(text)):
    if text[i].isalpha():
        countOfletters += 1
    elif text[i].isspace():
        countOfwords += 1
    elif text[i] == '.' or text[i] == '!' or text[i] == '?':
        countOfsentences += 1

L = countOfletters / countOfwords * 100
S = countOfsentences / countOfwords * 100
index = 0.0588 * L - 0.29 * S -  15.8
index = round(index)

if index > 16:
    print("Grade 16+")
if index > 1 and index < 17:
    print(f"Grade {index}")
else:
    print("Before Grade 1")
